### PR TITLE
Allow creation of multiple environments

### PR DESCRIPTION
**Pull request by Gaitholabi**
Adding variable named volumes in the `volumes` section of each service is incorrect since docker-compose will always refer to the volume by its defined name under compose. For example:

```
volumes:
    supportpal_mailer: #volume name reference in compose
        external: true
        name: ${MAILER_VOLUME} # volume real name in Docker
```

As shown in the documentation: https://docs.docker.com/compose/compose-file/compose-file-v3/#volume-configuration-reference

This change enables the user to actually have more than one environment running in parallel. Also, so that the compose-services aren't mixed up, the user should also duplicate the templates directory on the file system instead of initiating the two environments from the same compose-files.

### CODE REVIEW

**Comment by bytestream** [comment]
> This change enables the user to actually have more than one environment running in parallel.

I don't think this is true because the same issue occurs with service name - gateway already exists etc. I believe when container_name is defined docker doesn't generate a unique name and so it clashes.

**Comment by Gaitholabi** [comment]
> > This change enables the user to actually have more than one environment running in parallel.
> 
> I don't think this is true because the same issue occurs with service name - gateway already exists etc. I believe when container_name is defined docker doesn't generate a unique name and so it clashes.

yup, you still need need to change the other variables in the `.env` files. Namely, the 'container_name`

**Comment by bytestream** [comment]
That seems to cause further issues? Makefile uses this variables and it should be using the service name not the container name

**Comment by Gaitholabi** [comment]
> That seems to cause further issues? Makefile uses this variables and it should be using the service name not the container name

what issues did you face? the `service` name is internal, and you could refer to the container by its name

**Comment by bytestream** [comment]
`make install` errors saying no such service if you append `-test` to all service names

**Comment by Gaitholabi** [comment]
> `make install` errors saying no such service if you append `-test` to all service names

I just retested this

> `make install` errors saying no such service if you append `-test` to all service names

Although I wasn't able to replicate this,  but I found another issue with conflicting services and compose projects. Can you test after pulling?

**Comment by bytestream** [comment]
```
$ make install
cp -n .env.dist .env || true
cp -n docker-compose.yml.dist docker-compose.yml || true
cp -n docker-compose.prod.yml.dist docker-compose.prod.yml || true
cp -n -R ../../configs/gateway . || true
docker run -e SECRETS_DIR="/supportpal/helpdesk-install/templates/docker-compose/secrets/" -v "/supportpal/helpdesk-install/templates/docker-compose/secrets/:/secrets" "public.ecr.aws/supportpal/helpdesk-configurator:latest" sh "//app//scripts//create_secrets.sh"
Secrets generated successfully at /supportpal/helpdesk-install/templates/docker-compose/secrets/
docker volume create --name supportpal_db-test
supportpal_db-test
docker volume create --name supportpal_config-test
supportpal_config-test
docker volume create --name supportpal_storage-test
supportpal_storage-test
docker volume create --name supportpal_cache-test
supportpal_cache-test
docker volume create --name supportpal_mailer-test
supportpal_mailer-test
docker-compose -f docker-compose.yml -f docker-compose.prod.yml up -d gateway-test supportpal-test db-test redis-test
Creating volume "supportpal_supportpal_resources" with default driver
Creating volume "supportpal_supportpal_app" with default driver
ERROR: No such service: gateway-test
make: *** [Makefile:17: install] Error 1
```

```dotenv
COMPOSE_PROJECT_NAME=supportpal

# setup
DB_VOLUME=supportpal_db-test
CONFIG_VOLUME=supportpal_config-test
STORAGE_VOLUME=supportpal_storage-test
CACHE_VOLUME=supportpal_cache-test
MAILER_VOLUME=supportpal_mailer-test

# services
WEB_SERVICE_NAME=supportpal-test
MAILER_SERVICE_NAME=mailer-test
GATEWAY_SERVICE_NAME=gateway-test
DATABASE_SERVICE_NAME=db-test
CACHE_SERVICE_NAME=redis-test
CRON_SERVICE_NAME=supportpal_cron-test
```

**Comment by Gaitholabi** [comment]
> ```
> $ make install
> cp -n .env.dist .env || true
> cp -n docker-compose.yml.dist docker-compose.yml || true
> cp -n docker-compose.prod.yml.dist docker-compose.prod.yml || true
> cp -n -R ../../configs/gateway . || true
> docker run -e SECRETS_DIR="/supportpal/helpdesk-install/templates/docker-compose/secrets/" -v "/supportpal/helpdesk-install/templates/docker-compose/secrets/:/secrets" "public.ecr.aws/supportpal/helpdesk-configurator:latest" sh "//app//scripts//create_secrets.sh"
> Secrets generated successfully at /supportpal/helpdesk-install/templates/docker-compose/secrets/
> docker volume create --name supportpal_db-test
> supportpal_db-test
> docker volume create --name supportpal_config-test
> supportpal_config-test
> docker volume create --name supportpal_storage-test
> supportpal_storage-test
> docker volume create --name supportpal_cache-test
> supportpal_cache-test
> docker volume create --name supportpal_mailer-test
> supportpal_mailer-test
> docker-compose -f docker-compose.yml -f docker-compose.prod.yml up -d gateway-test supportpal-test db-test redis-test
> Creating volume "supportpal_supportpal_resources" with default driver
> Creating volume "supportpal_supportpal_app" with default driver
> ERROR: No such service: gateway-test
> make: *** [Makefile:17: install] Error 1
> ```
> 
> ```
> COMPOSE_PROJECT_NAME=supportpal
> 
> # setup
> DB_VOLUME=supportpal_db-test
> CONFIG_VOLUME=supportpal_config-test
> STORAGE_VOLUME=supportpal_storage-test
> CACHE_VOLUME=supportpal_cache-test
> MAILER_VOLUME=supportpal_mailer-test
> 
> # services
> WEB_SERVICE_NAME=supportpal-test
> MAILER_SERVICE_NAME=mailer-test
> GATEWAY_SERVICE_NAME=gateway-test
> DATABASE_SERVICE_NAME=db-test
> CACHE_SERVICE_NAME=redis-test
> CRON_SERVICE_NAME=supportpal_cron-test
> ```

I see. Looks like the issue is within the makefile, when we refer to the services via `docker-compose up {service_name}`
This doesn't accept the `container_name` as a reference.

So effectively,  we should refer  to everything tightly coupled to `compose` network with service name, but it's fine to use `container_name` with docker otherwise.

File: templates/docker-compose/docker-compose.prod.yml.dist [hunk 1]
```diff
@@ -10,8 +10,8 @@ services:
         volumes:
             - supportpal_resources:/supportpal/resources
             - supportpal_app:/supportpal/app
-            - '${CONFIG_VOLUME}:/supportpal/config/production'
-            - '${STORAGE_VOLUME}:/supportpal/storage'
+            - supportpal_cache:/supportpal/config/production
```
Comment (bytestream) [inline templates/docker-compose/docker-compose.prod.yml.dist:13]:
```suggestion
            - supportpal_config:/supportpal/config/production
```

File: templates/docker-compose/docker-compose.prod.yml.dist [hunk 2]
```diff
@@ -10,8 +10,8 @@ services:
...
+            - supportpal_db:/supportpal/storage
```
Comment (bytestream) [inline templates/docker-compose/docker-compose.prod.yml.dist:14]:
```suggestion
            - supportpal_storage:/supportpal/storage
```

File: templates/docker-compose/.env.dist [hunk 3]
```diff
@@ -1,3 +1,4 @@
+COMPOSE_PROJECT_NAME=production
```
Comment (bytestream) [inline templates/docker-compose/.env.dist:1]:
```suggestion
COMPOSE_PROJECT_NAME=supportpal
```

**Comment by bytestream** [comment]
Thanks for your help. This is working as intended now.

**Pull request merged**